src/Components/Recommendations/BookRec.js

```diff
@@ -0,0 +1,244 @@
+      sort based on priority and display the top n books
+
+*/
+
+import React, { useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import BookCards from './BookRecCards';
+
+// airtable configuration
+const Airtable = require('airtable');
+
+const airtableConfig = {
+  apiKey: process.env.REACT_APP_AIRTABLE_USER_KEY,
+  baseKey: process.env.REACT_APP_AIRTABLE_BASE_KEY,
+};
+
+const base = new Airtable({ apiKey: airtableConfig.apiKey })
+  .base(airtableConfig.baseKey);
+
+export default function RecCardsDisplay({
+  minAge, maxAge, minGrade, maxGrade, raceEthnicity,
+}) {
+  const [book, setBook] = useState([]);
+  const gradeList = ['0 to Pre-K', 'Kindergarten', '1st', '2nd', '3rd', '4th', '5th', '6th', '7th', '8th', '9th', '10th', '11th', '12th'];
+  const prioMap = new Map();
+  const [recList, setRecList] = useState([]);
+  // const [ageRange, setAgeRange] = useState([]);
+  // const [gradeRange, setGradeRange] = useState([]);
+  /*
+  const [ethnicity, setEthnicity] = useState([]); */
+  // const [identity, setIdentity] = useState();
+  // const [filteredRec, setFilteredRec] = useState([]);
+
+  // const getCards = () => {
+  //   base('Book').select({ view: 'Grid view' }).all()
+  //     .then((records) => {
+  //       setBook(records);
+  //     });
+  // };
+
+  // Instead of using props, we pull bookId from URL
+  const { bookId } = useParams();
+
+  // Airtable Filter implementation
+  // filterByFormula: `IF(${field} = {id},
+  // SEARCH("search for age_range", "id"), IF("", "", SEARCH("${fieldKeyword}", {${field}})))`
+  /*
+IF(
+        ${field} = {id},
+        SEARCH("search for age_range", "id"),
+        IF("", "", SEARCH("${fieldKeyword}", {${field}}))
+      )
+  */
+  // console.log({ minAge });
+  // console.log({ maxAge });
+  // console.log({ minGrade });
+  // console.log({ maxGrade });
+  // console.log({ raceEthnicity });
+
+  function RecFilter(field, fieldKeyword) {
+    // const map1 = new Map();
+    base('Book').select({
+      view: 'Grid view',
+    }).eachPage((records, fetchNextPage) => {
+      setBook(records);
+      records.forEach((record) => {
+        // do not check with the identical book
+        if (bookId !== record.get('id')) {
+          let priority = 0;
+          if ((minAge <= record.fields.age_max) && (maxAge >= record.fields.age_min)) {
+            priority += 2;
+          }
+
+          const minGradeIndex = gradeList.indexOf(minGrade, 0);
+          const maxGradeIndex = gradeList.indexOf(maxGrade, 0);
+          const minGradeIndexRecord = gradeList.indexOf(record.fields.grade_min, 0);
+          const maxGradeIndexRecord = gradeList.indexOf(record.fields.grade_max, 0);
+
+          if ((minGradeIndex <= maxGradeIndexRecord) && (maxGradeIndex >= minGradeIndexRecord)) {
+            priority += 2;
+          }
+
+          if (raceEthnicity && record.fields['race/ethnicity']) {
```

Comment:
I think the designers also wanted genre to be considered, so a very similar function to this should be run on the genre!